### PR TITLE
Clean up of original rust library

### PR DESCRIPTION
**Pull request by cflewis**
This PR cleans up the starting code we have in a couple of ways:
1. Fixes formatting to be the same as the official style guide.
2. Adds doctest examples for most stuff.
3. I ran `rustfmt` to keep the code clean.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -1,165 +1,174 @@
...
+    // An array of LED structs.
     pub led: Vec<LED>,
-    // A single button struct
-    pub button: Button
+    // A single button struct.
+    pub button: Button,
 }
 
 impl Tessel {
-    // Factory metho for Tessel
```
Comment (Frijol) [inline src/lib.rs:22]: Good idea to replace this comment with a more helpful one?

File: src/lib.rs [hunk 2]
```diff
@@ -1,165 +1,174 @@
...
+        // Create models for the four LEDs.
         let red_led = LED::new("red", "error");
         let amber_led = LED::new("amber", "wlan");
         let green_led = LED::new("green", "user1");
         let blue_led = LED::new("blue", "user2");
 
-        // Create the button
-        let button =  Button::new();
+        // Create the button.
+        let button = Button { value: false };
```
Comment (Frijol) [inline src/lib.rs]: Per https://github.com/tessel/t2-firmware/issues/107, we don't want to expose the button in the API after all.

File: src/lib.rs [hunk 3]
```diff
@@ -1,165 +1,174 @@
...
-    // Factory method for returning a new port struct
-    pub fn new(path: &'static str) -> Port {
-        Port {
-            socket_path: path
-        }
-    }
-}
-
-#[allow(dead_code)]
+// TODO: Figure out how to override the path so secretly so the example
```
Comment (Frijol) [inline src/lib.rs]: typo "so secretly so"

**Comment by Frijol** [comment]
Thanks, this is great! I'm a Rust beginner, so pinging @johnnyman727 or @flaki to take a second look.

File: src/lib.rs (hunk 2 above)
Comment (cflewis) [inline src/lib.rs]: Done.

File: src/lib.rs (hunk 1 above)
Comment (cflewis) [inline src/lib.rs:22]: Done.

File: src/lib.rs (hunk 3 above)
Comment (cflewis) [inline src/lib.rs]: Done.

**Comment by cflewis** [comment]
I've added a new test that actually verifies the LED file is being written to as expected.

**Comment by johnnyman727** [comment]
Fantastic work, @cflewis!

**Pull request merged**